From f59d3427396c321a03503e1abca6c8dac8f3902f Mon Sep 17 00:00:00 2001
From: Tomas Babej <tbabej@redhat.com>
Date: Wed, 11 Nov 2015 14:23:43 +0100
Subject: [PATCH] ipachangeconf: Add ability to preserve section case

The IPAChangeConf normallizes section names to lower case. There are
cases where this behaviour might not be desirable, so provide a way to
opt out.

https://fedorahosted.org/freeipa/ticket/5069

Reviewed-By: Gabe Alford <redhatrises@gmail.com>
---
 ipa-client/ipaclient/ipachangeconf.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ipa-client/ipaclient/ipachangeconf.py b/ipa-client/ipaclient/ipachangeconf.py
index 46132212112ff198e742f530860df4b15ea3312b..0c68a46e49f02bdb52fcd6e3683becca2965535f 100644
--- a/ipa-client/ipaclient/ipachangeconf.py
+++ b/ipa-client/ipaclient/ipachangeconf.py
@@ -59,6 +59,7 @@ class IPAChangeConf:
         self.deol = self.eol[0]
         self.sectnamdel = ("[", "]")
         self.subsectdel = ("{", "}")
+        self.case_insensitive_sections = True
 
     def setProgName(self, name):
         self.progname = name
@@ -110,7 +111,9 @@ class IPAChangeConf:
         return False
 
     def matchSection(self, line):
-        cl = "".join(line.strip().split()).lower()
+        cl = "".join(line.strip().split())
+        cl = cl.lower() if self.case_insensitive_sections else cl
+
         if len(self.sectnamdel) != 2:
             return False
         if not cl.startswith(self.sectnamdel[0]):
-- 
2.6.4

